app/assets/javascripts/modules/namespaces/components/new-form.js

```diff
@@ -65,40 +65,28 @@ export default {
       name: {
         required,
         format(value) {
-          // extracted from models/namespace.rb
-          const regexp = /^[a-z0-9]+(?:[._-][a-z0-9]+)*$/;
-
-          // required already taking care of this
-          if (value === '') {
-            return true;
-          }
-
-          return regexp.test(value);
-        },
-        available(value) {
           clearTimeout(this.timeout.name);
 
           // required already taking care of this
           if (value === '') {
-            return true;
+            return "";
           }
 
           return new Promise((resolve) => {
-            const searchName = () => {
-              const promise = NamespacesService.existsByName(value);
-
-              promise.then((exists) => {
-                // leave it for the back-end
-                if (exists === null) {
-                  resolve(true);
+            const validate = () => {
+              const promise = NamespacesService.validate(value);
+
+              promise.then((data) => {
+                if (data.exists) {
```

Comment:
Here we could just `set(this, 'errors', data.messages)` and then we can handle properly on the templates.

Reply:
Algo need to handle when `null` is returned.